### DISCUSSION TITLE
Question on "Contains_faces"

### DISCUSSION THREAD

**Discussion by giteeeeee** [TheLastBen/fast-stable-diffusion#374] [Q&A]
Could you explain what does this new feature do?

Thanks

**Comment by TheLastBen** [comment]
Simply it applies prior preservation, but only on the text_encoder, still experimental, so feedback from users would help.

**Reply by DarkAlchy** [reply]
I noticed that the old method is now gone.  I also noticed it is now 35% instead of 10% for the text encoding.  I trained a model at 4000 steps and 10% after a failure of 3000 and 15% (I couldn't style them very well).  Under the new system what would it be as 4k and 10% with faces male I have very mixed results.  Not very good looking model if I try and style them.

For 30 images what would be a good number, now, for steps and text with faces on?

**Reply by TheLastBen** [reply]
4k steps an 10% will allow easy styling, the results would be bad only if you're input images didn't contain enough information.

**Reply by DarkAlchy** [reply]
Is 4k steps and 10% enough to have a quality image?

**Reply by TheLastBen** [reply]
for 20 instance images, make it 5k 10%.
but it really depends on the input images, the lighting, the colors ... etc

**Reply by DarkAlchy** [reply]
I use 30 images though.

**Reply by TheLastBen** [reply]
make it 20 for this time

**Reply by DarkAlchy** [reply]
One thing I still do not get about resuming.  If I already trained with 10% then resume do I make that now 0% since it already trained 10%?

**Reply by TheLastBen** [reply]
just uncheck the box train_text_encoder, or set it to 0% it's the same thing

**Reply by DarkAlchy** [reply]
The result was absolutely horrid.  I described it below to help.  edit:  I looked at the notebook but I couldn't figure out where can we set the text at manually instead of the percentage?  We need a way to set that manually versus the percentage deal.  Maybe I was 270 step for text for a 5k steps of training but with the percentage it becomes a hassle and I am not even sure it would allow 5.4% in the example.

**Comment by TheLastBen** [comment]
Getting great results with it though  (1500 text_encoder, 3000 steps unet) : 

![new](https://user-images.githubusercontent.com/110583491/199878044-d840f540-1191-4323-9638-2009c9490bce.png)
![new2](https://user-images.githubusercontent.com/110583491/199878055-83adfa9c-dcc4-492d-b1cf-71fe89f260df.png)

**Reply by shyt47** [reply]
so its set to 3000 steps and 35% I guess for the text encoder. Would that get similar results as above or do I change that 35 to 1500?

**Reply by giteeeeee** [reply]
What's a unet? Is it just the name for the training method without text encoder?

**Reply by TheLastBen** [reply]
A model has many parts, the vae, tokenizer, unet and text_encoder

**Reply by shyt47** [reply]
> What's a unet? Is it just the name for the training method without text encoder?

Think he means unet=you get results like his examples above.

**Reply by DarkAlchy** [reply]
I took a step back and thanks to colab going to try TLB again.  With your example you are saying you had 4500 steps in total?  If so then this percentage thing is not really giving 1500 due to rounding.

Would it not be best to allow us to input how many steps instead of a percentage, or did you run for 1500 steps with the Train_text_encoder_for to 100 then when it is done change step count to 3k, uncheck the train text, check resume training then begin again?

**Reply by TheLastBen** [reply]
You can do it as you want, I personally train without the text_enc at the beginning, then set it to 100% and resume training for 1000s or more, finetuning the text_enc at the end seems to give better results.

**Reply by DarkAlchy** [reply]
> You can do it as you want, I personally train without the text_enc at the beginning, then set it to 100% and resume training for 1000s or more, finetuning the text_enc at the end seems to give better results.

I was thinking of doing that but then I never know how many to give to the unet.  Right now I did 1500 text then resumed with 3000 unet to see how this goes.  In your scenario how do we know when unet is done?

**Reply by TheLastBen** [reply]
depends on the training subject and the number of instance images, for example, I trained a model on 24 on 30 images, 768px, 7500 steps total with 1000s text_encoder, I got amazing results, it was trained on a photography style

**Reply by DarkAlchy** [reply]
My experiment was an utter disaster.  I trained 1500 of the text encoder and I could even make my model frown.  SHOCKED I am.  The problem is the face was crap.  I do 3k saving from 1500 every 500.  All crap even at 1500 where I lost my face and it became someone else.  POOF, under trained but had the overtraining pixelated stuff.  I wanted to really push it so What I did was train to 3k then to 5k saving every 250 steps.  I can say at 4k, and above, it was 100% unusable due to being overtrained.  Not using class images is a VERY big deal as on Shiv my models work so beautifully, but his has two issues.  It lacks fine-tuning as well as it now takes longer to encode than we get as a free colab.  Suddenly, as of yesterday, my 65m became 2h.  At 3.5k steps it is 3h and we only get 2.5h per day (which I just spent doing all this testing with TLB).

Having done this much testing I can see a few things and I will turn on the regularization images because it needs them, though they are not the same as I am used to we shall see.  I also switched to 3k then screw with the text encoder, but not being local I am so hampered.

Btw, why would an image go B&W then 500s still b&w, and another 500 steps b&w, then 500 more and colour from then on?  That shift to b&w while testing was weird since all before, and after, were in colour with each using the exact same settings for uniformity.

Doing your method is weird.  I trained the encoder for 1500 steps then trained the unet for 3k via resuming.  It just finished and did this for another 1500 steps.

![image](https://user-images.githubusercontent.com/10561690/201444307-2814aedb-f25e-4438-8a9f-6b811c6a4a2e.png)

Due to the above I ran out of time on colab and it yanked the GPU.  I have nothing to show for this experiment, and having to start all over would end up with the same outcome so can't be done by this free colab user as the 2.5h is just not long enough.

![image](https://user-images.githubusercontent.com/10561690/201444970-6daee1b5-ba22-4973-ae06-856c38379f83.png)

**Reply by TheLastBen** [reply]
once the text encoder is trained, when you resume, disable "train_text_encoder" to avoid training it again

**Reply by DarkAlchy** [reply]
> once the text encoder is trained, when you resume, disable "train_text_encoder" to avoid training it again

That is the point, I am not stupid, I had it unchecked.  If notice it trained for 1500 steps the text_encoder, then the 3k it trained for the unet then it finished and said it was training the unet again but for the 1500 steps I trained the text_encoder for the first round.

Started training today and started with 3200 unet YET it never says it is training unet.

![image](https://user-images.githubusercontent.com/10561690/201497857-e6b42916-77c1-4742-84c3-bbd19d659014.png)

![image](https://user-images.githubusercontent.com/10561690/201497869-50924aca-d9ef-4a5a-b848-4808831b8722.png)

**Comment by giteeeeee** [comment]
________________Old Method:                       prior preservation to instance images  +  prior preservation to text encoder
_______________New Method:                      ~~prior preservation to~~ instance images  +  ~~prior preservation to~~ text encoder
Experimental New Method : ~~prior preservation to~~ instance images  +  prior preservation to text encoder

Am I correct?

**Reply by TheLastBen** [reply]
Correct  👍

**Comment by TheLastBen** [comment]
@shyt47 set the total steps to 3000, text encoder to 35%, so the total would be (with the experimental new method) (35% of 3000) text encoder +3000 unet, so around 4500s

edit: sorry, around 4000 not 4500, the pictues above are 1k text_enc, 3k unet

**Comment by TheLastBen** [comment]
5000 unet 1000 text_enc : 

![download](https://user-images.githubusercontent.com/110583491/199891619-2e46697d-b83b-4921-8d4b-124b9848b324.png)

**Reply by shyt47** [reply]
Wow. Ok So I trained at 3000s with 35% So that was 1400. If I were to resume training what would I set both at for 5000 unet and 1000 text_ecn? Sorry for the dumb questions trying to wrap my simple mind around it.  I got pretty good results but needs a little fine tuning.

**Reply by TheLastBen** [reply]
If you resume training, disable text_encoder_training checkbox, you don't need to train the text_enc after training it 1400 steps

**Reply by shyt47** [reply]
ok so I would set the training to 2000 since I trained to 3000 previously?

**Reply by TheLastBen** [reply]
yes

**Reply by shyt47** [reply]
What is the prompt for above?

**Reply by TheLastBen** [reply]
_______,still frame by fred herzog, ultra detailed, beautiful, elegant, closeup

negative prompt :  cartoon, fake, painting, drawing, 3d, low poly

Steps: 50, Sampler: Euler a, CFG scale: 7, Size: 512x768, Denoising strength: 0.7, First pass size: 0x0, highres.fix

**Reply by MasterDenis** [reply]
> 5000 unet 1000 text_enc :
> 
> ![download](https://user-images.githubusercontent.com/110583491/199891619-2e46697d-b83b-4921-8d4b-124b9848b324.png)

how many images for this one?

**Reply by TheLastBen** [reply]
20

**Comment by giteeeeee** [comment]
In the normal new method, say setting 100% text encoder + 3000 training steps, is it just going to train 3000 steps of text encoder and 0 step unet? 

But in the experimental new method, if 100% text encoder + 3000 training steps, is it going to train 3000 steps of text encoder AND 3000 steps unet?

If so, how do I make it train only 3000 steps encoder but 0 step unet?

**Reply by TheLastBen** [reply]
Without training the unet, you won't see any resemblance to the trained subject

**Comment by TheLastBen** [comment]
in the normal new method if you set 100% text_enc, it will train both text_enc and unet at the same time

with the experimental, it can't train both at the same time because it uses regularization only for the text_enc, so 2 processes are needed for text_enc and unet.

I don't see a reason why you would train only the text_enc without training the unet, for 20 pics, you need only 350 steps of text_enc, adding also 350 to the unet won't hurt

**Reply by giteeeeee** [reply]
So in the normal new method, with 50% text_enc + 3000 steps, is to train 1500 steps text_enc AND unet, then 1500 steps of only unet?

**Reply by TheLastBen** [reply]
if you set the global step to 3000, 50% text_encoder, it will train 1500 text encoder + 3000 unet

**Reply by TheLastBen** [reply]
Because it doesn't train the unet during the first 1500 steps, and the unet needs a lot more steps than the text_enc

**Reply by giteeeeee** [reply]
> if you set the global step to 3000, 50% text_encoder, it will train 1500 text encoder + 3000 unet

3000 global steps, 50% text_encoder

Experimental: 1500 encoder, then 3000 unet

Normal: 1500 encoder+1500 unet, then 1500 unet

Is it right?

**Reply by TheLastBen** [reply]
in normal, when you set 3000, it will train both text_enc and unet for 3000 at the same time

**Reply by giteeeeee** [reply]
> in normal, when you set 3000, it will train both text_enc and unet for 3000 at the same time

In normal, text_enc and unet train at the same time

but when text_enc freezes at 50%/1500 step, then unet is trained alone for the rest of the steps?

**Comment by jx1k** [comment]
I trained a model yesterday with images of myself. Thirty images, Training_Steps set to 3000 and Contains_faces set to male. The results were comically bad with SD giving me giant lips and an extremely receding chin. I did another run today with Contains_faces set to no and the results have been good so far.
So am I right in assuming, that setting Contains_faces to No just does the training process like it used to be?

**Reply by TheLastBen** [reply]
yes, setting it to no sets the training to the normal one, how much did you set the text encoder when you did the training ?

**Reply by jx1k** [reply]
It was set to the default 35. The one I did today was at 40 because I had good results with that in the past.

**Reply by TheLastBen** [reply]
It is very important that the input images are high quality, the model doesn't see the images like a human, it only sees numbers, so even if the image looks okay doesn't mean it'll give you good results with training.

**Reply by jx1k** [reply]
Yes I understand, but it was only that particular training that went bad. I had a checkpoint from before, that was still trained with your old notebook (before the text encoder percentage was adjustable) and the identical set of images, which generated fantastic results. The one I trained today is also very good. I might give it another try tomorrow with the Contains_faces enabled just for testing, but as long as the "old" method without that setting still works I have no complaints.

**Reply by TheLastBen** [reply]
a lower % of the text_encoder reduces the weight of the instance, which means you'll need more steps, but allows style transfer easily

**Reply by jx1k** [reply]
Thanks, will keep experimenting!

**Reply by MasterDenis** [reply]
Would you Guys Say for specific people, say 21 images because the subject has complex face structure (and I wish to keep the face structure) would you do 50% encoder or perhaps less? I want to keep it balanced to avoid leakage whole also having good-enough accuracy of my subject without having the model aggressively print them on my final result, so far I got mixed results and because the colab is changing frequently it is hard to compare so I am looking for some headsup.

**Reply by TheLastBen** [reply]
to avoid leakage, use the option "contain_faces", set the text encoder to 30%, the steps from 5000 to 6000, and add another person to the training, use 21 pics, and another 20 pics of a different person (use a different identifier for them).
I noticed a better balance when training two people from different genders.

**Reply by TheLastBen** [reply]
Also, prompting is another important part, use this for testing : 

________, Still frame, cinematic, 1970s film, 40mm f/2.8,  4k uhd

negative prompt : cartoon, fake, painting

keep cfg scale at 7 and don't use many steps, 35 is enough, you can even go 15 or 25.

**Reply by TheLastBen** [reply]
experiment with two people from different genders in the same model (same seed different samplers, 35 steps) : 

![download (9)](https://user-images.githubusercontent.com/110583491/200029683-ee31ec4d-7e66-43a8-8cff-5ab4b43e5e32.png)

![download (10)](https://user-images.githubusercontent.com/110583491/200029707-9be7e755-325f-44a1-a494-68c2b4631f8e.png)

**Reply by MasterDenis** [reply]
> to avoid leakage, use the option "contain_faces", set the text encoder to 30%, the steps from 5000 to 6000, and add another person to the training, use 21 pics, and another 20 pics of a different person (use a different identifier for them). I noticed a better balance when training two people from different genders.

Wait, do you mean for **two** people training or **one** person?
I was curious about only one person training, specifically a woman with very complex face structure. I want to keep both face features but also face structure (cheeks and such) the issue was that their cheeks would over-exadurate and look like an old lady instead of a young woman.

PS: Could you implement a way to assign different classes to different set of images? (i.e. man, woman, dog, cat etc)
So in other words, perhaps say this token is a man, that token a woman. maybe to find a way so we can keep it cleaner.

**Reply by TheLastBen** [reply]
I meant for two people, If you use good quality input images you will certainly get good results, try with a celebrity to find the right settings, then train on other people, google has very good quality pics of celebs, make sure they are square, add imagesize:512x512 (or 1024x1024 ...etc) in google to get specific sizes.

**Reply by MasterDenis** [reply]
That is True, I did 21 Images 3k Steps 30% text encoder, I got good results overall but I asked the subject an they said it doesnt look like them (with the exact above prompt) I made a batch, some were distorted as expected of course slightly but their characteristics were indeed a bit too chubby, should I train more and if so how would you suggest to train them?
Thanks!

**Reply by MasterDenis** [reply]
> I meant for two people, If you use good quality input images you will certainly get good results, try with a celebrity to find the right settings, then train on other people, google has very good quality pics of celebs, make sure they are square, add imagesize:512x512 (or 1024x1024 ... etc) in google to get specific sizes.

Feedback: 
I did a few sessions, I tested using your prompt above, here are the results.
First I tried the subject with 3k Steps at 30% text encoder with 21 Images (Good Variety of Images with different backgrounds, clean light etc).
The Test had acceptable results (but not perfect, there were messy features sometimes), I achieved that with the CFG Scale at 7 - 7.5

Then I trained again, the same exact images, with 4k Steps and 25% text encoder, I achieved the same/similar results at about CFG Scale 8 or so (in terms of accuracy to make my subject 1:1 proportions etc).

Third was another and last training, I added +1000 steps without text encoder to the previous one, so it was 5k total steps with 1k steps (on the first session) for text encoder, in this case when I did the same configuration it would give me a completely different image of the subject (with the same seed) and it would bring back the same image as the previous sessions only at 8.5+ (anything below 8.5 would produce another image), I got an almost perfect replica of the subject at CFG Scale of 9 with this last session.

I hope this info helps with understanding the models better, I will keep you up to date for any other results, if you have any inputs about the above and how to possibly optimize it further, let me know!
I am in a real challenge since the subject is literarily complex as a girl's face.

**Reply by TheLastBen** [reply]
Find the right settings using pictures of a celeb, their pictures are taken by high end cameras

**Reply by MasterDenis** [reply]
> Find the right settings using pictures of a celeb, their pictures are taken by high end cameras

I see, I did some tests and so far with normal HD Mobile Phone Camera data (retouched and optimized for SD) my results were very mixed, 

Worst Results so far:
-training of Photos * 150 = training steps with 50% text encoder, I got the worst results with the eyes being very small and very distorted so I stopped at that.

Best Result:
 -(but not perfect 1:1 replica) 3000 steps with 31.7% text encoder with specifically 23 images (and num_class_images: 175).

Extra:
-With 4000 Steps (21 Images) and 25% text encoder (rest settings on default) I got Acceptable but a bit distorted results on some CFG Scales (I could not raise the CFG too much or it would look messy and very distorted with sharp lines all over the face), on the same session I pushed 1k Steps on Unet and it broke it, resulting in the person being unrecognizable slightly.
-Another Experiment with 2300 Steps at 35% text encoder (803 text encoder and 2300 unet) with 23 images (and num class images at default 200), in this session I started to notice those smaller and more distorted eyes while the overall head structure was stable-enough.
-Lastly I did an experiment with Photos*142 (21 Photos with 900 text_encoder steps and 3000 unet steps). On this session the results were acceptable when it comes to the head structure, almost perfect but the eyes were a bit distorted on some CFG Scales and the Head Structure was a bit more "fat" than normal (but this may be due to input data of the input images) on this session the subject's features were portrait well aside from the eyes.

The above was observed with the exact prompt and settings you suggested for the photo style to make accurate comparisons, I even made a CFG Scale comparison to see which produced the best likeness, sometimes I would get the subject 1:1 accurate on different scales but this effect can affect the workflow of prompt-editing, I guess the correct prompt is necessary across different models to optimize the results.

(All with "contains_faces" by the way)

I hope this feedback helps @TheLastBen 

I am still trying to optimize my settings, I run on an every-day woman with somewhat complex face structure (mixed race with some eastern and some western features so this is a challenge).

**Reply by DarkAlchy** [reply]
20 images at 5k steps and 10% text was absolutely horrid.  So many artefacts when generating, and I could not get it to pull to my prompt.  4k just about the same.  3k began to look better (less noise in the background and fewer artefacts on the model). 2k was so much better, but still hard to pull it due to the 500 steps (10%) of the text.  1k was absolutely the sweetest, but I lost facial features.

**Reply by MasterDenis** [reply]
> 20 images at 5k steps and 10% text was absolutely horrid. So many artefacts when generating, and I could not get it to pull to my prompt. 4k just about the same. 3k began to look better (less noise in the background and fewer artefacts on the model). 2k was so much better, but still hard to pull it due to the 500 steps (10%) of the text. 1k was absolutely the sweetest, but I lost facial features.

Maybe try out 1k with 20% (at least because he mentions at least 200 steps for text encoder)

**Reply by DarkAlchy** [reply]
Above he told me 5k, 20 images, and 10%  I tried it and ugh.

The main problem is getting the model to do as I say, but then the artefacts the higher I push the CFG to get it to do my bidding.  I think that is the text encoder.  This is why I mentioned to him above (in my edit) we need to stop with the percentage nonsense and allow us to do that or just input our own number irrespective of how many final steps we do.

**Reply by MasterDenis** [reply]
> Above he told me 5k, 20 images, and 10% I tried it and ugh.
> 
> The main problem is getting the model to do as I say, but then the artefacts the higher I push the CFG to get it to do my bidding. I think that is the text encoder. This is why I mentioned to him above (in my edit) we need to stop with the percentage nonsense and allow us to do that or just input our own number irrespective of how many final steps we do.

I know what you mean because you can just put the number of steps directly, now I have to every time calculate how much the % would be to subtract or add, anyhow, it is a little bit of a Hussle and I do not understand what is the visual difference.
Mine has also a lot of artefacts above lets say CFG 12 and that can cause issues when wanting to emphasize other things at the same time, it is actually a big issue but nothing that cant be fixed with massaging it with prompt editing, still though it would be nice if default settings served that purpose.

**Reply by TheLastBen** [reply]
keep the cfg at 7, more than that will ruin the results with dreambooth

**Reply by MasterDenis** [reply]
> Above he told me 5k, 20 images, and 10% I tried it and ugh.
> 
> The main problem is getting the model to do as I say, but then the artefacts the higher I push the CFG to get it to do my bidding. I think that is the text encoder. This is why I mentioned to him above (in my edit) we need to stop with the percentage nonsense and allow us to do that or just input our own number irrespective of how many final steps we do.

Based on observation I would say his Method makes it much easier to call my subject, however I have an issue with overfitting, its too easy to overfit compared to the pre-fork version, of course he tries to adjust it accordingly but it seems like its a bit aggressive, it strikes for likeness so far but the issue is that there is a very thin line.
So based on my observation its doing well at calling the subject at lower CFG, I was able to get my subject (almost) 1:1 at about below 7 CFG, thats both a good and a bad thing, if you want to prompt-engineer it, it will be difficult to call the subject as 1:1 AND have everything else around it be built in the way you want it without sacrificing something, the weights will be lower on ALL keywords and that could be an issue, ideally I want to have a good-enough likeness at around CFG 7-10, it depends of course.
I need to call my subject and have everything else be as it should.
Talking from a technical artist's perspective here, that would be "ideal".

So, Shiv's script has an issue with that as well, it makes it virtually impossible to bring the subject below 10 CFG.
Also I noticed that we got to be careful how much we train especially with Ben's script.

We have to remember that the users are most likely not coders so those things should be more clear of exactly what they affect and by how much so people can have a better idea of how to configure this.

Currently the 3000k steps 20 images and 31.7% text encoder seemed ideal but when I put more images with the same ratios (so 3500k steps and 31.7% text encoder) that caused a very very bad overfit at CFG 7, I got the most ideal results with:
3000 steps 31.7% text encoder with 23 images (and **class images at 175**).
This was Almost 1:1 (not quiet but close).
My goal was to have 1:1 likeness for photo-reproduction.
Perhaps also for restoration purposes (which is a bit more challenging considering there are not as many images and we have to improvise), so 1:1 is important in those use cases.

**Reply by TheLastBen** [reply]
what do you mean exactly by overfitting, is it leakage to other subjects or difficulty to apply style ?

**Reply by MasterDenis** [reply]
> what do you mean exactly by overfitting, is it leakage to other subjects or difficulty to apply style ?

to be more accurate the subject is starting to show as distorted, very distorted, the features will be meshed inwards and there will be artefacts in a triangle shape which are made out of the features of the subject, the higher CFG the more it happens, I had to go so far as 4-6 to get a balanced result, I did a series of experiments but on the one I mentioned I had an issue with not being able to rise my CFG without this issue.

**Reply by TheLastBen** [reply]
I never encountered such issue, it must be your input images

**Reply by DarkAlchy** [reply]
> > Above he told me 5k, 20 images, and 10% I tried it and ugh.
> > The main problem is getting the model to do as I say, but then the artefacts the higher I push the CFG to get it to do my bidding. I think that is the text encoder. This is why I mentioned to him above (in my edit) we need to stop with the percentage nonsense and allow us to do that or just input our own number irrespective of how many final steps we do.
> 
> Based on observation I would say his Method makes it much easier to call my subject, however I have an issue with overfitting, its too easy to overfit compared to the pre-fork version, of course he tries to adjust it accordingly but it seems like its a bit aggressive, it strikes for likeness so far but the issue is that there is a very thin line. So based on my observation its doing well at calling the subject at lower CFG, I was able to get my subject (almost) 1:1 at about below 7 CFG, thats both a good and a bad thing, if you want to prompt-engineer it, it will be difficult to call the subject as 1:1 AND have everything else around it be built in the way you want it without sacrificing something, the weights will be lower on ALL keywords and that could be an issue, ideally I want to have a good-enough likeness at around CFG 7-10, it depends of course. I need to call my subject and have everything else be as it should. Talking from a technical artist's perspective here, that would be "ideal".
> 
> So, Shiv's script has an issue with that as well, it makes it virtually impossible to bring the subject below 10 CFG. Also I noticed that we got to be careful how much we train especially with Ben's script.
> 
> We have to remember that the users are most likely not coders so those things should be more clear of exactly what they affect and by how much so people can have a better idea of how to configure this.
> 
> Currently the 3000k steps 20 images and 31.7% text encoder seemed ideal but when I put more images with the same ratios (so 3500k steps and 31.7% text encoder) that caused a very very bad overfit at CFG 7, I got the most ideal results with: 3000 steps 31.7% text encoder with 23 images (and **class images at 175**). This was Almost 1:1 (not quiet but close). My goal was to have 1:1 likeness for photo-reproduction. Perhaps also for restoration purposes (which is a bit more challenging considering there are not as many images and we have to improvise), so 1:1 is important in those use cases.

If I go above 7 it becomes ugly really fast with noise, and colour artefacts.  A lot of seeds seem to want me around CFG 4 then they look excellent, but, of course, at CFG 4 it isn't doing what I asked it to do.

**Reply by DarkAlchy** [reply]
> keep the cfg at 7, more than that will ruin the results with dreambooth

Dreambooth or your version of DB using the fast method?

**Reply by TheLastBen** [reply]
7 is working great from with the same settings you mentioned

**Reply by giteeeeee** [reply]
> > Above he told me 5k, 20 images, and 10% I tried it and ugh.
> > The main problem is getting the model to do as I say, but then the artefacts the higher I push the CFG to get it to do my bidding. I think that is the text encoder. This is why I mentioned to him above (in my edit) we need to stop with the percentage nonsense and allow us to do that or just input our own number irrespective of how many final steps we do.
> 
> Based on observation I would say his Method makes it much easier to call my subject, however I have an issue with overfitting, its too easy to overfit compared to the pre-fork version, of course he tries to adjust it accordingly but it seems like its a bit aggressive, it strikes for likeness so far but the issue is that there is a very thin line. So based on my observation its doing well at calling the subject at lower CFG, I was able to get my subject (almost) 1:1 at about below 7 CFG, thats both a good and a bad thing, if you want to prompt-engineer it, it will be difficult to call the subject as 1:1 AND have everything else around it be built in the way you want it without sacrificing something, the weights will be lower on ALL keywords and that could be an issue, ideally I want to have a good-enough likeness at around CFG 7-10, it depends of course. I need to call my subject and have everything else be as it should. Talking from a technical artist's perspective here, that would be "ideal".
> 
> So, Shiv's script has an issue with that as well, it makes it virtually impossible to bring the subject below 10 CFG. Also I noticed that we got to be careful how much we train especially with Ben's script.
> 
> We have to remember that the users are most likely not coders so those things should be more clear of exactly what they affect and by how much so people can have a better idea of how to configure this.
> 
> Currently the 3000k steps 20 images and 31.7% text encoder seemed ideal but when I put more images with the same ratios (so 3500k steps and 31.7% text encoder) that caused a very very bad overfit at CFG 7, I got the most ideal results with: 3000 steps 31.7% text encoder with 23 images (and **class images at 175**). This was Almost 1:1 (not quiet but close). My goal was to have 1:1 likeness for photo-reproduction. Perhaps also for restoration purposes (which is a bit more challenging considering there are not as many images and we have to improvise), so 1:1 is important in those use cases.

Have you experimented with the new method using 20 images? I've been using 30 images which might be too many?

**Reply by DarkAlchy** [reply]
[snip]
> Have you experimented with the new method using 20 images? I've been using 30 images which might be too many?

I did 20 images, 5k steps (saved every 1k steps) and 10% text encoding per Ben.  Not good.

**Reply by TheLastBen** [reply]
You can never find the best settings using instance images taken with average hardware, download high quality pictures from google of some known person, crop them with a good software and find the best settings, then apply those to your own instance pics. 99% of the problems are caused by the input pictures

**Reply by DarkAlchy** [reply]
Here is a pic of the training just calling it out using a good seed using the new DPM++2S Karras at 20 steps and CFG 7.
![image](https://user-images.githubusercontent.com/10561690/200176358-348115d8-b755-45d5-a789-c4f12387e191.png)

I caught a lucky seed but if I try and use the model to do something it begins to fall apart.  Either it ignores the request and looks alright or it does it and the artefacts/noise in the background is there.  Oh, let me add that is using the 2k step save as more than that it begins to look ugly no matter.  The 4k/5k is pretty bad.

**Reply by d1mski** [reply]
> 

I have come to the same conclusion. So far what I find to work well is using around 20-22 images, and training the text_enc for 30%. So in general as long as the unet is trained for approximately 2.5 times the steps used for the text_enc, things are working fairly well and I'm able to stylise my models with CFG set to 7. I had badly trained models but solely depending on input images. 

Not using contains_faces and on standard GPU (I believe it's a T4). Your INPUT images have to be of good quality in good light conditions. 

P.S. I will try to train the text_enc in the end as you suggested Ben and see if I get better results.

**Comment by pcaben** [comment]
I have a "interesting" experience with the contain face feature, I am a asian guy have long hair but most of the photo I use for training is me with ponytail hair, so should be not super outstanding.

 I choose men in contain face feature expected better training for men. In the final result that fine to create photo of me and the dress style just like the photo I use. But if I add more prompts for create painting e.g (fantasy, intricate, highly detailed, digital painting, artstation, concept art, smooth, sharp focus, illustration, art by artgerm and greg rutkowski and alphonse mucha). Something you standard painting prompts you can find in google.

The result will make me being a woman with breasts, I need to add extra prompts (man) to make myself be a man in the painting but also change my face a bit. lol...

**Reply by DarkAlchy** [reply]
Welcome to SD as that is one of its problem.  You can do a negative prompt and try it (woman) but for the most part this has nothing to do with TLB and everything to do with the diffusers and the images used when the master was trained.  If you look at those images there aren't a lot of Asians and not near as many men as there are women.  Sucks when you go to inpaint a face and it changes it from a man to woman, but it is what it is.

**Reply by bparrott78** [reply]
I re-ran the same pictures I had done with the old method through this new method and I can't say I really like it.  I tested the model every 500 steps and it just never spat out an image that was the likeness of the person being trained.  I had 20 face shots and 6 body shots from the front, back and sides.  With some VERY specific parameters and negatives I was able to get some decent results.  I have to say that the pictures themselves are good, but the face really doesn't match.

**Reply by TheLastBen** [reply]
@bparrott78 set the images resolution to 640 and train at 640, make sure the quality of the pictures is high, not a phone quality

**Reply by bparrott78** [reply]
Tried this, same result. Maybe I'm using too many pictures?
High quality pictures of beautiful women who look similar but definitely not the same.

**Reply by bparrott78** [reply]
Tested again, this time I ran the Shivam shirao notebook as well. Used the exact same images, saved every 500 to compare, trained on pictures of my wife.
My results were this: 
Shivam with prior preservation on at 2500 steps was the best representation of her. 
While the images that came from thelastben notebook were higher quality, the face seemed like it had been merged with other women. 
Going back and comparing against previous models I had trained, this was not the case.   This new method may be good for styles but I can't seem to get a good representation of a specific person.

**Reply by TheLastBen** [reply]
set the text encoder to 100%
